src/main/java/io/pravega/perf/PravegaPerfTest.java

```diff
@@ -468,11 +492,43 @@ public void shutdown(long endTime) {
         }
 
         public List<ReaderWorker> getConsumers() throws URISyntaxException {
+            return batchReaders ? getBatchConsumers() : getStreamingConsumers();
+        }
+
+        public List<ReaderWorker> getBatchConsumers() {
```

Comment:
A key difference between the batch and streaming readers in this benchmark is that there can be multiple benchmark processes reading together through a reader group. Adding more processes will increase the parallelism and data will not be read multiple times.
The batch reader does not perform this coordination so if one attempts to run multiple reader benchmark processes, each process will read the entire stream.
This behavior is reasonable but it should be documented.

Reply:
That's a good point and it was a subtly in the code that I missed, i.e. that the ReaderGroup name is devised from the steam name itself and therefore there is implicit coordination between multiple distributed processes.

https://github.com/pravega/pravega-benchmark/blob/master/src/main/java/io/pravega/perf/PravegaPerfTest.java#L290-L293
```
            if (recreate) {
                rdGrpName = streamName + startTime;
            } else {
                rdGrpName = streamName + "RdGrp";
            }
```
Obviously, as you point out, this is a bit harder to achieve with the BatchClient as the coordination is basically left up to the processor itself.  I'll document this current restriction and find a way of distributing the processing in another PR.

Reply:
If you need to distribute the processing (in another PR), consider accepting parameters for the process index and the number of processes. Then each process can sort the available segment ranges and take a deterministic subset.